core/lib/zksync_core/src/sync_layer/gossip/buffered/mod.rs

```diff
@@ -319,16 +315,32 @@ impl<T: ContiguousBlockStore> BlockStore for Buffered<T> {
 
 #[async_trait]
 impl<T: ContiguousBlockStore> WriteBlockStore for Buffered<T> {
-    async fn put_block(&self, ctx: &ctx::Ctx, block: &FinalBlock) -> StorageResult<()> {
+    /// Verify that `payload` is a correct proposal for the block `block_number`.
+    async fn verify_payload(
+        &self,
+        _ctx: &ctx::Ctx,
+        _block_number: BlockNumber,
+        _payload: &Payload,
+    ) -> ctx::Result<()> {
+        // This is storage for non-validator nodes (aka full nodes),
+        // so verify_payload() won't be called.
+        // Still, it probably would be better to either
+        // * move verify_payload() to BlockStore, so that Buffered can just forward the call
+        // * create another separate trait for verify_payload.
```

Comment:
IMO, it would be better to implement it as a separate trait (subtrait of `WriteBlockStore`, maybe) since this functionality indeed seems specific to validator nodes. OTOH, this could perform sanity checks and the hypothetical upstream caller in the consensus library would panic on an error (which would mean that the node cannot interpret the received block, or that the supermajority of validators has gone malicious).

Reply:
ack